src/api/spec/models/staging/staging_project_spec.rb

```diff
@@ -69,24 +71,55 @@
       it { expect(staging_project.overall_state).to eq(:unacceptable) }
     end
 
-    context 'when there are missing checks' do
+    context 'when there are missing checks on published repo' do
       before do
         repository.update(required_checks: ['check_1'])
       end
 
       it { expect(staging_project.overall_state).to eq(:testing) }
     end
 
+    context 'when there are missing checks on build repo' do
+      before do
+        repository_arch.update(required_checks: ['check_1'])
+      end
+
+      it { expect(staging_project.overall_state).to eq(:testing) }
+    end
+
     context 'when there are pending checks' do
-      let!(:check) { create(:check, name: 'check_1', status_report: status_report, state: 'pending') }
+      let!(:check) { create(:check, name: 'check_1', status_report: published_report, state: 'pending') }
```

Comment:
[nitpick]

All `let!(:check) { ... }` are used directly, so they should all be in a `before` block. Except for the last one, which could be used on line 118 for `required_check: [check.name]`

Reply:
why would you put the let! in a before block? The before is implicit

Reply:
I wasn't clear enough in what I meant. I meant that all the `let!(:check)` (except the last one) aren't needed since the variable `check` is never called. Instead, it should be like this:

```ruby
before do
  create(:check, name: 'check_1', status_report: published_report, state: 'pending')
end

it { expect(staging_project.overall_state).to eq(:testing) }
```

Using `let(:something)` is only when you need to call/use that `something` variable. Otherwise, it should be in a `before` block.

Reply:
> Otherwise, it should be in a before block.

Says who? :smile:

Reply:
Just think about it, why having a variable if you don't use it? I would say the common sense says it.

Reply:
before and let! are just two ways of doing

> force the method's invocation before each example.

See https://relishapp.com/rspec/rspec-core/docs/helper-methods/let-and-let

Reply:
But I guess it boils down to personal preference... this is why I put the [nitpick].

Reply:
https://dzone.com/articles/rspec-let-vs-before:

**In nearly every situation, it is better to use let over before blocks.**

Reply:
It's nice to omit the part which is exactly my point